File: src/Agrob4Simulation-ros-melodic-AgRob_simulator-launch-weta_v2/AgRob_simulator/launch/weta_v2/scripts/publisher.py
#!/usr/bin/env python

import rospy
from sensor_msgs.msg import PointCloud2
#from geometry_msgs.msg import PoseStamped

def callback(msg):
    modified_msg = PointCloud2()
    modified_msg.data = msg.data
    modified_msg.header = msg.header
    modified_msg.header.frame_id = "rslidar"  # Changing the frame ID
    

    # Additional modifications can be made here if necessary
    pub.publish(modified_msg)

if __name__ == '__main__':
    rospy.init_node('topic_modifier_node')
    original_topic = "/rslidar_points"
    
    sub = rospy.Subscriber(original_topic, PointCloud2, callback)
    pub = rospy.Publisher("/rslidar_points_modified", PointCloud2, queue_size=10)

    rospy.spin()

File: src/Agrob4Simulation-ros-melodic-AgRob_simulator-launch-weta_v2/AgRob_simulator/launch/weta_v2/publisher.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>
#include <geometry_msgs/PoseStamped.h>

ros::NodeHandle nh;
ros::Subscriber sub;
ros::Publisher pub;
std::string original_topic;

void callback(const geometry_msgs::PoseStamped::ConstPtr& msg) {
    geometry_msgs::PoseStamped modified_msg = *msg;
    modified_msg.header.frame_id = "rslidar"; 

    pub.publish(modified_msg);
}

int main(int argc, char** argv) {
    ros::init(argc, argv, "topic_modifier_node");
    original_topic = "/rslidar_points";
    
    sub = nh.subscribe(original_topic, 10, callback);
    pub = nh.advertise<geometry_msgs::PoseStamped>("/rslidar_modified", 10);

    ros::spin();

    return 0;
}
